packages/core/src/application.ts

```diff
@@ -133,6 +139,9 @@ export class Application extends Context {
   private _setupOperation(verb: string, path: string, spec: OperationObject) {
     const handler = spec['x-operation'];
     if (typeof handler === 'function') {
+      spec = Object.assign({}, spec);
```

Comment:
BTW I am going to rename `getApiSpec` to `getControllerSpec` to prevent confusion. The new name also makes more sense now that function is returning `ControllerSpec` type instead of `OpenApiSpec`.